spec/rest/tls.test.js

```diff
@@ -0,0 +1,118 @@
+"use strict";
+
+define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
+	var currentTime, exports = {};
+
+	/*
+	 * Set up and time-sync.
+	 */
+	exports.setupauth = function(test) {
+		test.expect(1);
+		helper.setupApp(function() {
+			var rest = helper.AblyRest();
+			(function(callback) {
+				rest.time(function(err, time) {
+					if(err) { callback(err); }
+					callback(null, time);
+				});
+			})(function(err, time) {
+				if(err) {
+					test.ok(false, helper.displayError(err));
+					test.done();
+					return;
+				}
+				currentTime = time;
+				test.ok(true, 'Obtained time');
+				test.done();
+			});
+		});
+	};
+
+	/*
+	 * tls:true initiates https connection
+	 */
+	exports.tls_connection = function(test) {
+		test.expect(4);
+
+		var rest,
+		keyStr = helper.getTestApp().keys[1].keyStr;
+
+		rest = helper.AblyRest({ tls: true, key: keyStr });
+
+		async.parallel([
+			function(cb) {
+				rest.auth.requestToken(function(err, tokenDetails) {
+					test.ok( tokenDetails.token, 'Verify that token is returned');
+					cb();
+				});
+			},
+			function(cb) {
+				rest.stats(function(err, stats) {
+					test.ok(err, 'Verify that an error was returned')
+					test.equal(err.code, 40160, 'Verify that error returned is "action not permitted" (from server)');
+					cb();
+				});
+			}
+		], function() {
+			test.equal(rest.baseUri().substr(0,5), "https", 'Requested token and stats fron tls endpoint');
+			test.done();
+		});
+	};
+
+	/*
+	 * tls:false initiates an http connection if authentication is token-based.
+	 */
+	exports.non_tls_connection = function(test) {
+		test.expect(3);
+
+		var rest = helper.AblyRest({ tls: false, useTokenAuth: true });
+
+		async.parallel([
+			function(cb) {
+				rest.auth.requestToken(function(err, tokenDetails) {
+					test.ok( tokenDetails.token, 'Verify that token is returned');
+					cb(err);
+				});
+			},
+			function(cb) {
+				rest.stats(function(err, stats) {
+					test.ok(stats, 'Verify that stats are returned');
+					cb(err);
+				});
+			}
+		], function(err) {
+			if(err)
+				test.ok(false, "Request returned an error: " + helper.displayError(err));
+			test.equal(rest.baseUri().substr(0,5), "http:", 'Requested token and stats fron non-tls endpoint');
+			test.done();
+		});
+	};
+
+	/*
+	 *  The combination of tls:false and basic auth results in error (see RSC18, RSA1).
+	 *  The error MUST NOT occur at instantiation, because we might need the Rest client 
+	 *  for things that don't require sending a key. The error MUST occur on any request 
```

Comment:
Ah, that's a fair point, although I am not sure why anyone would explicitly ask for a client to not use TLS only to check the time, or for that matter to create a signed token request. Based on @SimonWoolf's feedback though leave as is, sorry for the mixed messages.